packages/relay/src/lib/eth.ts

```diff
@@ -601,12 +605,22 @@ export class EthImpl implements Eth {
         this.logger.info(`${requestIdPrefix} Returning gas: ${response.result}`);
         return prepend0x(trimPrecedingZeros(response.result));
       } else {
+        this.logger.error(`${requestIdPrefix} No gas estimate returned from mirror-node: ${JSON.stringify(response)}`);
         return this.predefinedGasForTransaction(transaction, requestIdPrefix);
       }
     } catch (e: any) {
       this.logger.error(
         `${requestIdPrefix} Error raised while fetching estimateGas from mirror-node: ${JSON.stringify(e)}`,
       );
+      // in case of contract revert, we don't want to return a predefined gas but the actual error with the reason
+      if (
+        this.estimateGasThrows &&
+        e instanceof MirrorNodeClientError &&
+        e.isContractReverted() &&
+        !this.estimateGasIgnoredErrors.includes(e.message)
```

Comment:
If `e.isContractReverted()` is `true`, can the error message be `from field must not be empty`? Just to make sure we have right guard here.

Reply:
Oh, I see, thanks for the clarification.

Reply:
I added a new method inside `MirrorNodeClientError`:
```typescript
  public isContractRevertOpcodeExecuted() {
    return this.message === MirrorNodeClientError.messages.CONTRACT_REVERT_EXECUTED;
  }
```

which is taken from [here](https://github.com/hashgraph/hedera-sdk-js/blob/507fdad9a4ef399b929a8dd1a167c8619109f8b5/src/Status.js#L111) and used that instead of checking the message inside the `this.estimateGasIgnoredErrors` array. 

There are too many validations done inside the mirror node and we need to add each type of validation inside this array in order to avoid bugs - I saw some dApp tests were failing because they were passing wrong transaction bodies to `eth_estimateGas` and they were receiving an error instead of getting the predefined gas.